quinn/src/streams.rs

```diff
@@ -149,6 +151,39 @@ where
         conn.wake();
     }
 
+    /// Completes if/when the peer stops the stream, yielding the error code
+    pub fn stopped(&mut self) -> Stopped<'_, S> {
+        Stopped { stream: self }
+    }
+
+    #[doc(hidden)]
+    pub fn poll_stopped(&mut self, cx: &mut Context) -> Poll<Result<VarInt, WriteError>> {
+        let mut conn = self.conn.lock().unwrap();
```

Comment:
I think this could reasonably be a 0-RTT stream, in which case we should probably handle the possibility of 0-RTT being rejected and emit a `ZeroRttRejected` error, similar to `finish`. That way a client attempting 0-RTT doesn't and up inadvertently waiting for an unrelated stream to be stopped.